objectKeys/challenges-06.test.js

```diff
@@ -0,0 +1,304 @@
+// values from the object.
+// ------------------------------------------------------------------------------------------------
+
+const getValues = (obj) => {
+  return Object.values(obj);
+}
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 3
+//
+// Write a function named getEntries that takes in an object and returns an array of the entries
+// (key/value pairs) from the object.
+// ------------------------------------------------------------------------------------------------
+
+const getEntries = (obj) => {
+  return Object.entries(obj);
+}
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 4
+//
+// Write a function named getFrom that takes in an object and a property name and returns the
+// information from the object. For example, if invoked with "courseInfo" and "keys" as arguments,
+// it should return an array containing the keys for the courseInfo object
+// (['name', 'duration', 'topics', 'finalExam']).
+//
+// This will make our code more dynamic and DRY.
+// ------------------------------------------------------------------------------------------------
+const courseInfo = {
+  name: 'Code 301', duration: { dayTrack: '4 weeks', eveningTrack: '8 weeks' },
+  topics: ['SMACSS', 'APIs', 'NodeJS', 'SQL', 'jQuery', 'functional programming'],
+  finalExam: true
+};
+
+const getFrom = (obj, property) => {
+  return Object[property](obj);
+}
+
+// ------------------------------------------------------------------------------------------------
+// CHALLENGE 5
+//
+// Use the characters data below for the rest of the challenges.
+//
+// Write a function named totalCharacters that takes in an array and returns the number of
+// characters in the array. Use the getFrom function you wrote in challenge 4.
+// ------------------------------------------------------------------------------------------------
+
+let characters = [
+  {
+    name: 'Eddard',
+    spouse: 'Catelyn',
+    children: ['Robb', 'Sansa', 'Arya', 'Bran', 'Rickon'],
+    house: 'Stark'
+  },
+  {
+    name: 'Jon',
+    spouse: 'Lysa',
+    children: ['Robin'],
+    house: 'Arryn'
+  },
+  {
+    name: 'Cersei',
+    spouse: 'Robert',
+    children: ['Joffrey', 'Myrcella', 'Tommen'],
+    house: 'Lannister'
+  },
+  {
+    name: 'Daenarys',
+    spouse: 'Khal Drogo',
+    children: ['Drogon', 'Rhaegal', 'Viserion'],
+    house: 'Targaryen'
+  },
+  {
+    name: 'Mace',
+    spouse: 'Alerie',
+    children: ['Margaery', 'Loras'],
+    house: 'Tyrell'
+  },
+  {
+    name: 'Sansa',
+    spouse: 'Tyrion',
+    children: [],
+    house: 'Stark'
+  },
+  {
+    name: 'Jon',
+    spouse: null,
+    children: [],
+    house: 'Snow'
+  }
+]
+
+const totalCharacters = (arr) => {
+  const newArr = [];
+  arr.forEach(obj => {
+    newArr.push(getFrom(obj, 'keys'));
+  });
+  return newArr.length;
+}
```

Comment:
keys() returns an array so you simply need `return getFrom(arr, 'keys').length;`